x/oracle/types/message_proposals.go

```diff
@@ -0,0 +1,181 @@
+	}
+	return nil
+}
+
+func NewMsgRemoveAssetInfo(authority, denom string) *MsgRemoveAssetInfo {
+	return &MsgRemoveAssetInfo{
+		Authority: authority,
+		Denom:     denom,
+	}
+}
+
+// Implements Msg Interface
+var _ sdk.Msg = &MsgRemoveAssetInfo{}
+
+func (msg *MsgRemoveAssetInfo) Route() string {
+	return RouterKey
+}
+
+func (msg *MsgRemoveAssetInfo) Type() string {
+	return TypeMsgAddAssetInfo
+}
+
+func (msg *MsgRemoveAssetInfo) GetSigners() []sdk.AccAddress {
+	creator, err := sdk.AccAddressFromBech32(msg.Authority)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{creator}
+}
+
+func (msg *MsgRemoveAssetInfo) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func (msg *MsgRemoveAssetInfo) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(msg.Authority)
+	if err != nil {
+		return errorsmod.Wrapf(sdkerrors.ErrInvalidAddress, "invalid feeder address (%s)", err)
+	}
+	return nil
+}
+
+// NewMsgAddPriceFeeders creates a new MsgAddPriceFeeders instance
+func NewMsgAddPriceFeeders(
+	authority string,
+	feeders []string,
+) *MsgAddPriceFeeders {
+	return &MsgAddPriceFeeders{
+		Authority: authority,
+		Feeders:   feeders,
+	}
+}
+
+func (msg *MsgAddPriceFeeders) Route() string {
+	return RouterKey
+}
+
+func (msg *MsgAddPriceFeeders) Type() string {
+	return TypeMsgAddAssetInfo
+}
+
+func (msg *MsgAddPriceFeeders) GetSigners() []sdk.AccAddress {
+	creator, err := sdk.AccAddressFromBech32(msg.Authority)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{creator}
+}
+
+func (msg *MsgAddPriceFeeders) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func (msg *MsgAddPriceFeeders) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(msg.Authority)
+	if err != nil {
+		return errorsmod.Wrapf(sdkerrors.ErrInvalidAddress, "invalid feeder address (%s)", err)
+	}
+	return nil
+}
+
+func NewMsgRemovePriceFeeders(authority string, feeders []string) *MsgRemovePriceFeeders {
+	return &MsgRemovePriceFeeders{
+		Authority: authority,
+		Feeders:   feeders,
+	}
+}
+
+// Implements Msg Interface
+var _ sdk.Msg = &MsgRemovePriceFeeders{}
+
+func (msg *MsgRemovePriceFeeders) Route() string {
+	return RouterKey
+}
+
+func (msg *MsgRemovePriceFeeders) Type() string {
+	return TypeMsgAddAssetInfo
+}
+
+func (msg *MsgRemovePriceFeeders) GetSigners() []sdk.AccAddress {
+	creator, err := sdk.AccAddressFromBech32(msg.Authority)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{creator}
+}
+
+func (msg *MsgRemovePriceFeeders) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func (msg *MsgRemovePriceFeeders) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(msg.Authority)
+	if err != nil {
+		return errorsmod.Wrapf(sdkerrors.ErrInvalidAddress, "invalid feeder address (%s)", err)
+	}
+	return nil
+}
```

Comment:
@jelysn those changes causes the gov proposals prior to 114 to fail to display as some proposals are using old proposal structure.

Reply:
fixing this by removing old proposals